### DISCUSSION TITLE
search Doesn't work

### DISCUSSION THREAD

**Discussion by fadellabanie** [livewire/livewire#1719] [4. General]
i have route 
web\routes
Route::get('/properties', Properties::class)->name('properties');

 code in view livewire
<input wire:model="search" type="text" class="form-control" id="kt_subheader_search_form" placeholder="Search...">

livewire controller 
class Properties extends Component
{ 
    use WithPagination;
   
    public $search = '';
    public $page = 1;
    protected $updatesQueryString = ['search'];

    public function mount(): void
    {
        $this->fill(request()->only('search', 'page'));
    }
    public function paginationView()
    {
        return 'custom-pagination-links-view';
    }
    public function render()
    {
        return view('livewire.properties', [
            'properties' => $this->search === null ?
            Property::paginate(12) :
            Property::where('name_', 'like', '%' . $this->search . '%')->paginate(12)
        ]) ->extends('admin.layouts.app')
        ->section('content');
    }

when Move between pages search input doesn't work 
it's work only when refresh page

**Comment by iksaku** [comment]
Your search logic may need to be place in a separate function.

If you use [Livewire's Computed Properties](https://laravel-livewire.com/docs/2.x/properties#computed-properties), you can solve this issue.

Having something like:
```php
public function getPropertiesProperty()
{
    // Your search + Pagination logic here...
}
```

Then, in your blade component, access the property using `$this->properties`, that will make reactivity work better ;)
